copis/core.py

```diff
@@ -34,14 +34,19 @@
 from queue import Empty as QueueEmpty
 from queue import Queue
 from typing import List, Optional, Tuple
+from glm import vec3
 
 from pydispatch import dispatcher
 
 import copis.coms.serial_controller as serial_controller
 
 from .enums import ActionType, DebugEnv
 from .helpers import Point5
-from .classes import Action, Device, MonitoredList
+from .classes import (
+    Action, Device, MonitoredList, Object3D, CylinderObject3D, AABBObject3D,
+    OBJObject3D)
+
+MAX_ID = 16777214

```

Comment:
We "pick" objects in the viewport (cameras, action points, and now proxy objects) by doing a second "picking pass". In the picking pass, we render all objects we want to be able to select as a single color corresponding to its id. It's sort of a hack, but for our purposes it's perfect (you can read more about it [here](http://www.opengl-tutorial.org/miscellaneous/clicking-on-objects/picking-with-an-opengl-hack/)). Each color channel is 0-255 so the max id representable is 0xFFFFFF for white. 

0xFFFFFF (16777215) is reserved because we use it for the background, so the max id that our program can render with the picking method is 16777214. If there are d devices, a actions/points, and p proxy objects, devices get ids [0, d), actions/points get ids [d, d+a), and i've made it so that proxy objects get ids [16777215 - p - 1, 16777215). Hence why we have `MAX_ID` as 16777214.

`MAX_ID` would optimally be kept in a globals file that can define constants like these. For now i've just inserted it wherever we need to use it, which I admit isn't great. We could for now move it into `enums.py`? I think we could also rename `enums.py` to globals or something else, whatever you think.

Reply:
yeah, move it to `globals.py`. And we can put other global constants in there too in the future.

Reply:
Done!